docs/docs/tutorials/deployment/index.md

```diff
@@ -54,14 +56,45 @@ async def predict(question: Question):
         }
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
+
+# Define an endpoint (streaming)
+from fastapi.responses import StreamingResponse
+
+@app.post("/predict/stream")
+async def stream(question: Question):
+    async def generate():
+        async for value in streaming_dspy_program(question=question.text):
+            if isinstance(value, dspy.Prediction):
+                data = {"prediction": value.labels().toDict()}
+            elif isinstance(value, litellm.ModelResponse):
+                data = {"chunk": value.json()}
+            yield f"data: {ujson.dumps(data)}\n\n"
+        yield "data: [DONE]\n\n"
+
+    return StreamingResponse(generate(), media_type="text/event-stream")
+
+# Since you're often going to want to stream the result of a DSPy program as server-sent events,
+# we've included a helper function for that, which is equivalent to the code above.
+
+from dspy.utils.streaming import streaming_response
+
+@app.post("/predict/stream")
+async def stream(question: Question):
+    stream = streaming_dspy_program(question=question.text)
+    return StreamingResponse(streaming_response(stream), media_type="text/event-stream")
```

Comment:
Awesome example! Not sure if you have bandwidth, but it would be great to introduce test coverage like this as well by:

1. Extending https://github.com/stanfordnlp/dspy/blob/main/tests/test_utils/server/litellm_server.py to introduce a streaming endpoint

2. Writing a test case that uses `litellm_test_server` + streaming, similar to https://github.com/stanfordnlp/dspy/blob/97ca14cdae61b14bb1c2d520ba559e710d2cb7f6/tests/clients/test_lm.py#L9

Reply:
@dbczumar would love your help on this, I'm stuck on the `miprov2-knn` branch for benchmarking atm

Reply:
Done :)